targets/pna_nic/pna_nic.h

```diff
@@ -79,6 +79,7 @@ class PnaNic : public Switch {
 
  private:
   static packet_id_t packet_id;
+  static constexpr port_t PNA_PORT_RECIRCULATE = 0xfffffffa;
```

Comment:
There is no mention of a recirculation port in the PNA spec. @jafingerhut is this correct?
All I know is that this is the SDN port value defined by the P4Runtime spec as the recirculation port.
I guess targets are free to "choose" (or define) the value used for the recirculation port.

Reply:
Yes, there is a recirculate() operation in PNA, and if that is done, then the packet should recirculate when it completes the current execution of the main control.  There is NO recirculation port like there is in PSA or v1model architectures.

Reply:
Note: As an implementation technique, it is probably a reasonable idea to have a C++ boolean or 1-bit flag somewhere associated with the packet in the PNA implementation that is initialized to 0/false whenever a packet begins execution of the main control, and it is set to 1/true when the recirculate() extern function is called.

Reply:
I couldn't find a `recirculate()` extern function in [pna.p4](https://github.com/p4lang/pna/blob/main/pna.p4)? Is there a reference to it somehere.
If PNA specifies such an extern function, then this is what this PR should introduce (not a special port value). Implementation of the extern function can use a special port value if appropriate (the boolean flag approach you suggested may be a bit more difficult within the constraints of bmv2).

Reply:
Yes. I didn't find any `recirculate()` extern function in [pna.p4](https://github.com/p4lang/pna/blob/main/pna.p4). So, I followed the PSA implementation.

If introducing a new `recirculate()` function is better approach, we can implement that.

Reply:
Hi @antoninbas, Is the current approach Ok? Or should I change it recirculate() extern approach?

Reply:
One approach is to wait until there is an official PNA definition for how recirculation is done, and behaves, and implement other features that are already well-defined.

Reply:
Ok